packages/thirdweb/src/extensions/pack/createPack.test.ts

```diff
@@ -0,0 +1,187 @@
+import { describe, expect, it } from "vitest";
+import { ANVIL_CHAIN } from "~test/chains.js";
+import { TEST_CONTRACT_URI } from "~test/ipfs-uris.js";
+import { TEST_CLIENT } from "~test/test-clients.js";
+import { TEST_ACCOUNT_A } from "~test/test-wallets.js";
+import { getContract } from "../../contract/contract.js";
+import { download } from "../../storage/download.js";
+import { sendAndConfirmTransaction } from "../../transaction/actions/send-and-confirm-transaction.js";
+import { approve } from "../erc20/write/approve.js";
+import { mintTo as mintToERC20 } from "../erc20/write/mintTo.js";
+import { setApprovalForAll } from "../erc721/__generated__/IERC721A/write/setApprovalForAll.js";
+import { mintTo as mintToERC721 } from "../erc721/write/mintTo.js";
+import { getPackContents } from "../erc1155/__generated__/IPack/read/getPackContents.js";
+import { getTokenCountOfBundle } from "../erc1155/__generated__/IPack/read/getTokenCountOfBundle.js";
+import { getUriOfBundle } from "../erc1155/__generated__/IPack/read/getUriOfBundle.js";
+import { openPack } from "../erc1155/__generated__/IPack/write/openPack.js";
+import { deployERC20Contract } from "../prebuilts/deploy-erc20.js";
+import { deployERC721Contract } from "../prebuilts/deploy-erc721.js";
+import { deployPackContract } from "../prebuilts/deploy-pack.js";
+import { createPack } from "./createPack.js";
+
+const account = TEST_ACCOUNT_A;
+const client = TEST_CLIENT;
+const chain = ANVIL_CHAIN;
+
+describe.runIf(process.env.TW_SECRET_KEY)("createPack", () => {
+  it("should work", async () => {
```

Comment:
better messages please, remember ppl come behind you to fix things when this breaks :) leave the house tidy